specification/security_model.adoc

```diff
@@ -1,9 +1,242 @@
+
+* _Privileged guest software adversary_: (Other TVM guest kernels)
+
+* _Device firmware adversary_: (Other system firmware, DMA into unassigned TVMs)
+
+* _Hardware adversary_: This includes adversaries that have unlimited physical
+  access to the system and its devices, and can use simple or advanced hardware
+  attacks such as bus interposers, voltage/clock glitchers, microscopes, power
+  usage analyzer, etc. in order to access or tamper with the protected assets.
+
+=== Threats
+
+==== CoVE-IO-T001
+
+.CoVE-IO-T001
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Disclosure
+| Privileged host software
+| In scope
+| Host component reads TVM confidential data
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can program direct memory access destination addresses to its own address
+    space. Device generated data intended to be copied to the TVM is accessed by
+    the host component instead.
+
+|===
+
+==== CoVE-IO-T002
+
+.CoVE-IO-T002
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Tamper
+| Privileged host software
+| In scope
+| Host component writes to TVM confidential memory
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can modify the programmed direct memory access destination addresses, and
+    redirect it to unintended parts of the TVM address space.
+
+|===
+
+==== CoVE-IO-T003
+
+.CoVE-IO-T003
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Denial of service
+| Privileged host software
+| **Not** in scope
+| TVM can not access a TDI that is assigned to it
+
+5+| **Description** - A privileged host software component resets or powers down
+    an assigned TDI or its physical device, while the TDI is assigned to a TVM.
+    The TVM is no longer able to directly access its assigned TDI.
+
+|===
+
+==== CoVE-IO-T004
+
+.CoVE-IO-T004
+|===
+
+| Asset | Threat | Adversary | Scope | Result
+
+| Device trusted MMIO
+| Tamper
+| Privileged host software
+| In scope
+| TVM programs a TDI that is unassigned to it
+
+5+| **Description** - A privileged host software component remaps a TVM assigned
+    TDI MMIO guest physical address to an unassigned TDI MMIO host physical
+    address. The TVM programs a different TDI than the one that is assigned to
+    it.
+
+|===
+
+==== CoVE-IO-T005
+
+.CoVE-IO-T005
+|===
+
+| Asset | Threat | Adversary | Scope | Result
+
+| Device trusted MMIO
+| Denial of service
+| Privileged host software
+| In scope
+| TVM can not access the trusted MMIO space of a TDI that is assigned to it
+
+5+| **Description** - A privileged host software component configures PCIe
+    switches to redirect or drop MMIO accesses from the TVM to one of its
+    assigned TDIs.
```

Comment:
Is this really in scope?

The VMM can touch the device and make IDE into insecure state **any time**.

I am not sure how that can be mitigated...

Reply:
I changed that one to be about the host VMM redirecting MMIO access from a TVM to an untrusted device or to another, unassigned device. This can be mitigated with the mandatory usage of selective stream and the protection of the RP IDE stream registers.

Reply:
Yes, I am OK to change the threat to "tamper".

I just feel uncomfortable to see "denial of service", because it is almost impossible to mitigate.